lib/compiler.js

```diff
@@ -634,9 +673,33 @@ function buildAllMixinInstructions(appRootDir, mixinDirs, options) {
       _.extend(meta, require(metafile));
     }
     meta.sourceFile = filepath;
-    return meta;
+    mixinInstructions[meta.name] = meta;
   });
 
+  return _.values(mixinInstructions);
+}
+
+function fetchMixinNamesUsedInModelInstructions(modelInstructions) {
+  return _.flatten(modelInstructions
+  .map(function(model) {
+    return model.definition && model.definition.mixins ?
+      Object.keys(model.definition.mixins) : [];
+  }));
+}
+
+function filterMixinInstructionsUsingWhitelist(instructions, includeMixins) {
+  return instructions
+    .filter(function(m) {
+      return (includeMixins.indexOf(m.name) === -1 ? false : true);
+    });
+}
+
+function concatMixinInstructions(rejectionSet, inclusionSet) {
+  var mixins = _.reject(rejectionSet, function(instruction) {
+    return (_.findWhere(inclusionSet,
+      {name: instruction.name}) !== undefined);
+  });
+  mixins = mixins.concat(inclusionSet);
```

Comment:
This function is pretty complicated and difficult to understand. I am proposing the following change:

 - modify `loadMixins` to return a hashmap (lookup table), not an array
 - replace `concatMixinInstructions` with

   ```js
_.assign(instructionsFromMixinSources, instructionsFromMixinDirs, instructionsFromMixins)
```

Thoughts?